src/ripple/conditions/RsaSha256.h

```diff
@@ -0,0 +1,302 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONDITIONS_RSA_SHA256_H
+#define RIPPLE_CONDITIONS_RSA_SHA256_H
+
+#include <ripple/conditions/Condition.h>
+#include <ripple/conditions/Fulfillment.h>
+#include <ripple/conditions/impl/base64.h>
+#include <ripple/protocol/PublicKey.h>
+#include <ripple/protocol/SecretKey.h>
+#include <ed25519-donna/ed25519.h>
+#include <openssl/bn.h>
+#include <openssl/evp.h>
+#include <openssl/rsa.h>
+#include <openssl/pem.h>
+#include <openssl/sha.h>
+
+#include <boost/algorithm/clamp.hpp>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <algorithm>
+#include <iterator>
+
+namespace ripple {
+namespace cryptoconditions {
+
+class RsaSha256 final
+    : public Fulfillment
+{
+    struct rsa_deleter
+    {
+        void operator() (RSA* rsa) const
+        {
+            RSA_free (rsa);
+        }
+    };
+
+    using RsaKey = std::unique_ptr<RSA, rsa_deleter>;
+
+    struct bn_deleter
+    {
+        void operator() (BIGNUM* bn) const
+        {
+            BN_free (bn);
+        }
+    };
+
+    using BigNum = std::unique_ptr<BIGNUM, bn_deleter>;
+
+    Buffer modulus_;
+    Buffer signature_;
+
+    // Check whether the public modulus meets the length
+    // requirements imposed by section 4.4.1 of the RFC.
+    static
+    bool
+    checkModulusLength (int len)
+    {
+        if (len <= 0)
```

Comment:
Probably not, but I'd rather keep it since the input is the result of `OpenSSL` which, often, work in mysterious ways. It's insurance on the cheap.